trieste/models/gpflux/interface.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+import tensorflow as tf
+from gpflow.base import Module
+
+from ...types import TensorType
+from ..interfaces import ProbabilisticModel
+
+
+class GPfluxPredictor(ProbabilisticModel, tf.Module, ABC):
+    """A trainable wrapper for a GPflux deep Gaussian process model. The code assumes subclasses
+    will use the Keras `fit` method for training, and so they should provide access to both a
+    `model_keras` and `model_gpflux`. Note: due to Keras integration, the user should remember to
+    use `tf.keras.backend.set_floatx()` with the desired value (consistent with GPflow) to avoid
+    dtype errors."""
```

Comment:
It makes me uneasy

Reply:
This is a difficult issue, but it's more on the GPflux side. To use Keras (which our models assume we will) the user will have to set `tf.keras.backend.set_floatx()`. From a discussion I had with Hrvoje he felt it was best to have the user do this, rather than us doing it for the user, in case the user wants to move away from float64

Reply:
I agree with the idea of having user set this, otherwise it would a side effect that is hard to identify and prevent. We just need to be very loud about that in our docs, preferably in multiple places.

Reply:
Actually, is a runtime check we could do, and perhaps output a warning if float64 isn't set?

Reply:
I was worried a bit that this would interfere with the rest of the toolbox so thought to leave it for the users - we point it out in the notebook, but you are right, ti should be stressed in multiple places, added to the docstrings etc

good point of a runtime check @apaleyes

Reply:
After discussing with Andrei, we've decided this is annoyingly difficult to have a runtime check for (GPflow uses 'np.float64' whereas Keras uses 'float64' for instance and it's not obvious how to compare the two). So we're just going to make sure it's well-documented.